lldb/source/Plugins/ExpressionParser/Clang/ClangUserExpression.cpp

```diff
@@ -669,15 +670,8 @@ bool ClangUserExpression::Parse(DiagnosticManager &diagnostic_manager,
   // Parse the expression
   //
 
-  Process *process = exe_ctx.GetProcessPtr();
-  ExecutionContextScope *exe_scope = process;
-
-  if (!exe_scope)
-    exe_scope = exe_ctx.GetTargetPtr();
-
-  bool parse_success = TryParse(diagnostic_manager, exe_scope, exe_ctx,
-                                execution_policy, keep_result_in_memory,
-                                generate_debug_info);
+  bool parse_success = TryParse(diagnostic_manager, exe_ctx, execution_policy,
+                                keep_result_in_memory, generate_debug_info);
```

Comment:
Per above comment this would just become:
```
bool parse_success = TryParse(diagnostic_manager, exe_ctx.GetBestExecutionContextScope(), execution_policy, keep_result_in_memory, generate_debug_info);
```

Reply:
Removing `exe_ctx` in favour of `exe_scope` is trickier because we store away the `exe_ctx` for use in `ClangExpressionDeclMap`. I'd prefer sticking with this approach unless people are strongly opposed to it

Reply:
At some point we should go through and rationalize the use of ExecutionContext vrs. ExecutionContextScope, but this patch doesn't seem the place to do that.  Instead, going with whatever is locally most convenient seems the right choice here.

Reply:
@clayborg Any remaining concerns with landing this as-is? It seems like we want to have a separate discussion around which one of `ExecutionContextScope` or `ExecutionContext` to use in APIs, orthogonal to this patch.

Reply:
@clayborg ping